server/image_pull.go

```diff
@@ -66,6 +70,10 @@ func (s *Server) PullImage(ctx context.Context, req *types.PullImageRequest) (*t
 		s.pullOperationsLock.Lock()
 		defer s.pullOperationsLock.Unlock()
 		pullOp, inProgress = s.pullOperationsInProgress[pullArgs]
+		if named, err := reference.ParseNormalizedNamed(pullArgs.image); err == nil {
```

Comment:
The normalization to `docker.io` can lead to false positives. `fedora` and `library/fedora` would both resolve to `docker.io/library/fedora` although `fedora` has an alias to `registry.fedoraproject.org`.

I would really prefer we'd use `s.pullOperationsInProgress`.  Both have corner cases but using the existing struct, the corner cases would be consistent.